apps/nestjs-wallet/docs/rebeam-tutorial.md

```diff
@@ -0,0 +1,511 @@
+    "verificationMethod": [
+        {
+            "id": "did:key:z6MkrMzkYMnPvQptgPk3E9BmkNzYGFh1Hs5U3WZHTDJgufvn#z6MkrMzkYMnPvQptgPk3E9BmkNzYGFh1Hs5U3WZHTDJgufvn",
+            "type": "Ed25519VerificationKey2018",
+            "controller": "did:key:z6MkrMzkYMnPvQptgPk3E9BmkNzYGFh1Hs5U3WZHTDJgufvn",
+            "publicKeyJwk": {
+                "crv": "Ed25519",
+                "x": "sPMLz2IQhMA-_W2urtbiUyEEgEf_kgRpZfnZXDGXvLc",
+                "kty": "OKP",
+                "kid": "dT4nPSrLt_Vngf0Bpo0ZanOQ0HcduNB8AM-_4KKk_30"
+            }
+        }
+    ]
+}
+```
+
+#### [Supplier] Issue "customer contract" credential
+
+First, the supplier needs a DID from which they can issue a credential.
+Again, navigate to the `DID Controller create` request under the `did` folder.
+Ensure the request body is as shown and send.
+```json
+{
+    "method": "key"
+}
+```
+
+Again, this should give a response similar to this one, with a different `did`.
+Note down the `id` property. This is the suppliers's DID.
+
+```json
+{
+    "id": "did:key:z6MkqtPi8BfTpC1gitAePJz73A1aTj5gEeecQHW4pwfC9PeJ",
+    "verificationMethod": [
+        {
+            "id": "did:key:z6MkqtPi8BfTpC1gitAePJz73A1aTj5gEeecQHW4pwfC9PeJ#z6MkqtPi8BfTpC1gitAePJz73A1aTj5gEeecQHW4pwfC9PeJ",
+            "type": "Ed25519VerificationKey2018",
+            "controller": "did:key:z6MkqtPi8BfTpC1gitAePJz73A1aTj5gEeecQHW4pwfC9PeJ",
+            "publicKeyJwk": {
+                "crv": "Ed25519",
+                "x": "qeC-F4pWhck7tIplhNo-qkISlfa2_H2x0fVBNqH0XDs",
+                "kty": "OKP",
+                "kid": "96gNvxG0qgZVYsmuME2EZfH3EwaWvtRalzKOEYw4Y5E"
+            }
+        }
+    ]
+}
+```
+
+After having created a new DID, the supplier can then issue a credential to the customer DID that was previously created.
+Typically, a supplier would confirm that the customer in question controls the DID first, however this step is currently omitted.
+
+Navigate to the `Vc Api Controller issue Credential` request under the `vc-api` folder.
+Fill in, in the JSON below, the Customer DID as the `subject` id, the Supplier DID as the `issuer` id and the `verificationMethod.id` from the DID document of the Supplier as the `options.verificationMethod` from the DIDs that were generated in previous steps.
+
+```json
+{
+  "credential": {
+        "@context":[
+            "https://www.w3.org/2018/credentials/v1",
+            {
+                "issuerFields":{
+                    "@id":"ew:issuerFields",
+                    "@type":"ew:IssuerFields"
+                },
+                "namespace":"ew:namespace",
+                "role":{
+                    "@id":"ew:role",
+                    "@type":"ew:Role"
+                },
+                "ew":"https://energyweb.org/ld-context-2022#",
+                "version":"ew:version",
+                "EWFRole":"ew:EWFRole"
+            }
+        ],
+        "id":"urn:uuid:7f94d397-3e70-4a43-945e-1a13069e636f",
```

Comment:
Maybe we can note that this is a uuid v4 and should be different per request

Reply:
Hmm, because we shouldn't issue the same credential twice you mean?